lib/elm_common/BugzillaLogin.elm

```diff
@@ -0,0 +1,197 @@
+update msg model =
+  case msg of
+    Logged credentials ->
+      let
+        model_ = { model | credentials = credentials}
+      in
+        -- Check credentials when received from storage
+        (model_, checkCredentials model_ False)
+
+    UpdateLogin login ->
+      let
+        newCreds = case model.credentials of
+          Just creds -> { creds | login = login }
+          Nothing -> { login = login, token = ""}
+      in
+        ({ model | credentials = Just newCreds}, Cmd.none)
+      
+    UpdateToken token ->
+      let
+        newCreds = case model.credentials of
+          Just creds -> { creds | token = token }
+          Nothing -> { login = "" , token = token}
+      in
+        ({ model | credentials = Just newCreds}, Cmd.none)
+      
+    SaveCredentials ->
+      -- Check credentials before saving them
+      (model, checkCredentials model True)
+
+    CheckedCredentials save check ->
+      (
+        { model | check = check },
+
+        -- Save credentials stored in model when valid
+        case check of
+          Success check' ->
+            if check' && save then
+              case model.credentials of
+                Just creds ->
+                  bugzillalogin_set creds
+                Nothing ->
+                  Cmd.none
+            else
+              Cmd.none
+          _ ->
+            Cmd.none
+      )
+
+checkCredentials : Model -> Bool -> Cmd Msg
+checkCredentials model save =
+  -- Check bugzilla auth is still valid
+  case model.credentials of
+    Just creds_ ->
+      let
+        task = buildRequest creds_ model.url "GET" 
+          (Http.url "/valid_login" [
+            ("login", creds_.login)
+          ])
+          Nothing -- no body
+      in
+        (Http.fromJson JsonDecode.bool task)
+        |> RemoteData.asCmd
+        |> Cmd.map (CheckedCredentials save)
+
+    Nothing ->
+      Cmd.none
+
+buildRequest: Credentials -> String -> String -> String -> Maybe String -> Task Http.RawError Http.Response
+buildRequest creds url verb method body =
+  let
+    body' = case body of
+      Just b -> Http.string b
+      Nothing -> Http.empty
+
+    headers = [
+      -- Use bugzilla token
+      ( "x-bugzilla-api-key", creds.token ),
+      ( "Accept", "application/json" ),
+      ( "Content-Type", "application/json" )
+    ]
+  in
+    -- Always send to Mozilla Bugzilla
+    Http.send Http.defaultSettings {
+      url = url ++ "/rest" ++ method,
+      verb = verb,
+      headers = headers,
+      body = body'
+    }
+
+
+view: Model -> Html Msg
+view user =
+  div [id "bugzilla", class "container"] [
+    h1 [] [text "Manage your Bugzilla credentials"],
+    viewEditor user,
+    hr [] [],
+    div [class "alert alert-info"] [
+      strong [] [text "Heads up !"],
+      p [] [text "Your Bugzilla credentials are stored only in your browser, using localStorage. They are NOT sent to any backend for storage."]
+    ]
+  ]
+
+viewEditor: Model -> Html Msg
+viewEditor user =
```

Comment:
``BugzillaLogin.elm`` should not enforce any html structure. more reusable would be to provide those tree messages as separate functions which return Html attribute

- ``onSubmit SaveCredentials``
- ``onInput UpdateLogin``
- ``onInput UpdateToken``